compiler/rustc_hir_typeck/src/method/mod.rs

```diff
@@ -182,8 +183,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         self_expr: &'tcx hir::Expr<'tcx>,
         args: &'tcx [hir::Expr<'tcx>],
     ) -> Result<MethodCallee<'tcx>, MethodError<'tcx>> {
-        let pick =
-            self.lookup_probe(segment.ident, self_ty, call_expr, ProbeScope::TraitsInScope)?;
+        let pick = self.lookup_probe(
+            segment.ident,
+            self_ty,
+            call_expr,
+            segment.res.opt_def_id(),
+            ProbeScope::TraitsInScope,
```

Comment:
Then this can be changed into like:

```
if let Some(only_method) = segment.res.opt_def_id() { ProbeScope::Single(only_method) } else { ProbeScope::TraitsInScope }
```